src/main/java/org/elasticsearch/script/ScriptService.java

```diff
@@ -553,8 +557,8 @@ public static void writeTo(ScriptType scriptType, StreamOutput out) throws IOExc
         public final String lang;
         public final String script;
 
-        public CacheKey(String lang, String script) {
-            this.lang = lang;
+        public CacheKey(ScriptEngineService scriptEngineService, String script) {
```

Comment:
yea it could. the reason to pass the service though is that we need to make sure that we always use the same lang acronym in case the engine supports more (think of `js` and `javascript` that are both supported and relate to the same engine). Allowing to pass in a string would make it possible to mess this up a bit, as this multiple types thing is a bit trappy. Thoughts?